### PR TITLE
fix(mobile): Sorted shared album and added share user doesn't reflect change in album view

### LINKED ISSUE #1954: [BUG] Mobile - Add a share user to an album doesn't reflect the change immediately
### The bug

This is only applicable to the current development branch

Adding a shared user to an album does not reflect the change on mobile. The user has to navigate to a different tab and navigate back to see the changes.

### The OS that Immich Server is running on

iOS

### Version of Immich Server

v1.50.1

### Version of Immich Mobile App

v1.50.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1. Go to Library Page
2. Go to an album
3. Click on add user -> select a user to add -> Add
4. Album doesn't show the shared user until you pop back, navigate to a different tab, and finally come back to the Library Page and tap on that album to see the changes.
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alextran1502**
Fix #1954

### CODE REVIEW

File: mobile/lib/modules/album/views/album_viewer_page.dart [hunk 1]
```diff
@@ -91,6 +91,8 @@ class AlbumViewerPage extends HookConsumerWidget {
             .addAdditionalUserToAlbum(sharedUserIds, album);
 
         if (isSuccess) {
+          // TODO - update state to reflect changes here
```
Comment (alextran1502) [inline mobile/lib/modules/album/views/album_viewer_page.dart:94]: @zoodyy Do you know why even though we invalidated the provider, the state doesn't update here?

Comment (zoodyy) [reply mobile/lib/modules/album/views/album_viewer_page.dart]: Yeah, figured it out and added a fix to this branch / PR:

`addAdditionalUserToAlbum` in `AlbumService` did not update the `shared` property of the album - only the shared user list. However, the UI won't show the shared users at all if `shared == false`. This now works for me.

While debugging I found a related server issue: The server does not update the `updatedAt` value of an album if a shared user is added or removed. This still needs to be fixed.

Comment (zoodyy) [reply mobile/lib/modules/album/views/album_viewer_page.dart]: Same issue with adding/removing assets to/from an album. server does not update `updatedAt` for the album

Comment (zoodyy) [reply mobile/lib/modules/album/views/album_viewer_page.dart]: fixed the server not updating `updatedAt` with #1977

**Comment by alextran1502** [comment]
@zoodyy, @martyfuhry If you don't have anything else to add in this PR, can you help me approve this one so I can merge it in?

**Review by fyfrey — APPROVED** [review]
@alextran1502 
Looks good to me since the required server changes have been merged

**Pull request merged**